src/main/java/net/mcreator/ui/ide/RSyntaxTextAreaStyler.java

```diff
@@ -58,7 +58,8 @@ public static void style(RSyntaxTextArea te, RTextScrollPane sp, int initialFont
 				theme = Theme.load(te.getClass().getResourceAsStream(
 						"/org/fife/ui/rsyntaxtextarea/themes/" + PreferencesManager.PREFERENCES.ide.editorTheme
 								.toLowerCase(Locale.ENGLISH) + ".xml"));
-			} else {
+			} else if (ThemeLoader.CURRENT_THEME.getID().equals("default_dark") ||

```

Comment:
This is expected. Themes can override MAIN_TINT to achieve this so please rever this part of this PR.

Reply:
I added themeId checking to allow custom *xml*s to specify **separate** colors for each type of reserved words (my theme does that, for example).
Is this not meant to be allowed? Or did <kbd>revert this part</kbd> mean removing the `else if` thing?

Reply:
If this was your plan, I think the best solution would be to make a special token that is replaced with MAIN_TINT inside xml file, eg. `${mainTint}` that is replaced here by main tint. This way themes can either use custom colors or main tint

Reply:
`code_editor.xml` is not a freemarker template, so I should add **either** a "code word" like `#main_tint` to the xml **or** a new parameter to `theme.json` that declares whether MAIN_TINT needs to be used or not (the first variant is better though).
I'll decide which option to implement after I'm done with #1439.

Reply:
I don't think you need freemarker for that, just use eg. replace("${mainTint}", maintint.tovalidformat())